crypto/x509/x_ietfatt.c

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright 2021 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/err.h>
+#include <openssl/asn1t.h>
+#include <openssl/x509_acert.h>
+
+/*-
+ * Definition of IetfAttrSyntax from RFC 5755 4.4
+ *
+ * IetfAttrSyntax ::= SEQUENCE {
+ *   policyAuthority [0] GeneralNames    OPTIONAL,
+ *   values          SEQUENCE OF CHOICE {
+ *                     octets    OCTET STRING,
+ *                     oid       OBJECT IDENTIFIER,
+ *                     string    UTF8String
+ *                   }
+ * }
+ *
+ * Section 4.4.2 states that all values in the sequence MUST use the
+ * same choice of value (octet, oid or string).  This restriction is
+ * not explicitly enforced by the current implementation.
```

Comment:
IMO, this should be enforced.

Reply:
@hlandau I think that it's easy enough to enforce this from the openssl API, but what about for importing these attributes from certificates generated elsewhere?  Do we need a  `d2i_OSSL_IETF_ATTR_SYNTAX()` implementation that returns `NULL` if the sequence value choices don't match?

Reply:
I've implemented this as I described here.

Reply:
@hlandau Did you have any comments about how the IetfAttrSyntax enforcement has been addressed here?